src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1190,18 +1192,24 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 setIcon(icon);
 
             } else if (value instanceof Long) {
```

Comment:
You could change this to `value instanceof Long len` so that the cast below can be eliminated.

Reply:
After replacing "value instanceof Long" by "value instanceof Long len" and removing the cast below (long len = value) results in build error. So, I guess cast is required to convert an object to long type.

Reply:
You don't need to use `value` in this `if` statement any longer. Just use `len` inside (that's the whole purpose).